storage/types.go

```diff
@@ -0,0 +1,52 @@
+package storage
+
+import (
+	"io"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+)
+
+// Storage represents the permanent storage abstraction
+// for reading and writing operations
+type Storage interface {
+	io.Closer
+	Reader
+	Writer
+}
+
+// StorageRead defines the transaction storage interface for read methods
```

Comment:
Nitpick: the comment doesn't match the interface name, probably leftover

Reply:
Resolved in [1a82f41](https://github.com/gnolang/tx-indexer/pull/8/commits/1a82f4175f3bbc267786ab215e4b69760c998958)